plugin/pkg/admission/serviceplan/defaultserviceplan/admission.go

```diff
@@ -86,13 +91,34 @@ func (d *defaultServicePlan) Admit(a admission.Attributes) error {
 		glog.V(4).Info(msg)
 		return admission.NewForbidden(a, errors.New(msg))
 	}
-	if len(sc.Plans) > 1 {
+	// find all the service plans that belong to the service class
+
+	// TODO: Need to be careful here. Is it possible to have only one
+	// serviceplan available while others are still in progress?
+
+	// implement field selector
+
+	// loop over all service plans accumulate into slice
+	plans, err := d.spLister.List(labels.Everything())
+	// TODO filter `plans` down to only those owned by `sc`.
```

Comment:
I am thinking on reading this that we should put field selection into this PR.

Reply:
Did you want to reply to this one?

Reply:
https://github.com/kubernetes-incubator/service-catalog/pull/1106#issuecomment-328223610